src/front/wgsl/mod.rs

```diff
@@ -705,9 +709,22 @@ impl<'a> ExpressionContext<'a, '_, '_> {
             arguments: self.arguments,
             block: self.block,
             emitter: self.emitter,
+            rhs: self.rhs,
         }
     }
 
+    fn reborrow_rhs(&mut self) -> ExpressionContext<'a, '_, '_> {
+        let mut reborrow = self.reborrow();
+        reborrow.rhs = true;
+        reborrow
+    }
+
+    fn reborrow_lhs(&mut self) -> ExpressionContext<'a, '_, '_> {
```

Comment:
I feel rather strongly that we shouldn't have all 3 of `reborrow()`, `reborrow_rhs()` and `reborrow_lhs()`. First, because the old `reborrow()` becomes rather exceptional. It's almost never that we really want to inherit this flag.
Second, because most of reborrows should be "rhs", so it should be the default.

How do you feel about:
  - `reborrow()` having the RHS semantics. Most of the usage stays unchanged therefore.
  - `reborrow_lhs` having the LHS semantics.
  - `reborrow_lhs_inherit` for inheriting the LHS from the parent, if any

Reply:
For me it's fine, I'll change it

Reply:
Actually, after trying to implement it I think this adds more confusion and is more error prone for example let's say I have this snippet

```rust
fn test(p: ptr<function, Struct>) {
   (*p).field = 1;
}
```

This is valid wgsl and for it to work in `(*p).field` it starts as lhs trough `parse_singular_expression` it then must go trough `parse_primary_expression` still as lhs, as it's a expression in parentheses it will go trough `parse_general_expression` it then goes trough 2 more calls until reaching `parse_singular_expression` where we finally see the deref, in all of this the lhs must be kept in context, it's more error prone in my opinion to know where it must be inherited than where it must cleared.

Reply:
there is a a very limited set of things that need to inherit the LHS status, it's precisely the "postfix" things and nothing else.